app/src/main/cpp/src/3rd_party/CMakeLists.txt

```diff
@@ -181,6 +181,28 @@ if(ENABLE_LOG AND NOT ANDROID)
   endif()
 endif()
 
+if(ANDROID)
+  set(OPENSSL_SOURCE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/openssl_src)
+  set(OPENSSL_INCLUDE_DIRECTORY ${3RD_PARTY_INSTALL_PREFIX}/include)
+  set(OPENSSL_LIBS_DIRECTORY ${3RD_PARTY_INSTALL_PREFIX}/lib)
+  set(OPENSSL_FOR_ANDROID_DIR ${3RD_PARTY_SOURCE_DIRECTORY}/android-helpers/openssl)
+
+  include(ExternalProject)
+  ExternalProject_Add(
+   openssl_external
+    URL https://www.openssl.org/source/openssl-1.1.1.tar.gz
+    DOWNLOAD_DIR ${OPENSSL_SOURCE_DIRECTORY}
+    SOURCE_DIR ${OPENSSL_SOURCE_DIRECTORY}
+    CONFIGURE_COMMAND cp ${OPENSSL_FOR_ANDROID_DIR}/build_openssl.sh ${OPENSSL_SOURCE_DIRECTORY}/build_openssl.sh &&
+                      cd ${OPENSSL_SOURCE_DIRECTORY} &&
+                      sh build_openssl.sh ${CMAKE_ANDROID_NDK} ./ ${3RD_PARTY_INSTALL_PREFIX} ${CMAKE_ANDROID_ARCH_ABI}
```

Comment:
@VadymLuchko I believe these two lines can be moved to BUILD_COMMAND?

Reply:
done in 048ae59c19ca4e61f504361cf27bd9190b2e5830